src/registry/RLNRegistry.ts

```diff
@@ -0,0 +1,25 @@
+import { IncrementalMerkleTree } from "@zk-kit/incremental-merkle-tree";
+import { Member } from "./types";
+import Registry from "./index";
+
+export default class RLNRegistry {
+  private _registry: Registry;
+  private _slashedRegistry: Registry;
+
+  /**
+   * Initializes the group with the tree depth and the zero value.
+   * @param treeDepth Tree depth.
+   * @param zeroValue Zero values for zeroes.
+   * @param hasSlashedRegistry Boolean flag to determine whether to create a SlashedRegistry.
+   */
+  constructor(
+    treeDepth = 20,
+    zeroValue: Member = BigInt(0),
+    hasSlashedRegistry = false
+  ) {
+    this._registry = new Registry(treeDepth, zeroValue);
+    this._slashedRegistry = hasSlashedRegistry
```

Comment:
I added _slashedRegistry property in case there's in need of prevent spammer from rejoining.
If it isn't appropriate for this protocol I'll remove this. 
Please leave a comment.

Reply:
If it's ok to add this property, I'll add function that if a member is slashed from _registry, then added to _slashedRegistry.

Reply:
Yes, that is great; having a reference to the slashed members is important in case they try to join again.

I would rename `this._slashedRegistry` to `this._slashed` and `hasSlashedRegistry` to `isSlashed` just for conciseness.